src/net/udp.rs

```diff
@@ -220,6 +220,11 @@ impl UdpSocket {
         self.inner.send_zc(buf).await
     }
 
+    /// Sends a message on the socket using a msghdr.
+    pub async fn sendmsg_zc(&self, msghdr: &libc::msghdr) -> (libc::msghdr, io::Result<usize>) {
```

Comment:
I believe the convention is to pass the size result first and the buffer ownership back second.